### PR TITLE
Lab2 1

### PR DESCRIPTION
**Pull request by tobeja**
AddEdge function should be done, let me know if you guys can't run it for some reason.

### CODE REVIEW

**Comment by kmfarley11** [comment]
It seems to run ok, but I noticed in the tester we have output such as this:
"ERROR: Edge must be between two nodes"

This could be fine; however, it LOOKS like an error. If you are expecting an error out of some of your prints / operations while testing, then please explicitly state this so that it is easier to understand the functionality which is tested and working. For all I know it is just telling me something bad happened which wouldn't be a practical thing to merge into master ;)

In other words please enhance your tests to specify what is being tested and what is working. Indentations and newlines also help to indicate differences in tests.

File: Lab2/digraph.cpp [hunk 1]
```diff
@@ -135,7 +135,82 @@ void Digraph::AddEdge(int taskOne, int taskTwo)
...
+
+  //check if starting node = target node
+  if (taskOne == taskTwo){
+		cout << "ERROR: Edge must be between two nodes\n";
+		return;
+  }
+
+  //traverse header list
+  while (ptr != NULL && ptr->taskNo != taskOne){
+	 if (ptr->taskNo == taskTwo)
```
Comment (kmfarley11) [inline Lab2/digraph.cpp]: Awkward indentations make code harder to read. In order to help me review your code please fix spacings/ indentations to increase readability. I don't mind different spacing than mine; however, your code should be more uniform looking.

File: Lab2/digraph.cpp~ [hunk 2]
```diff
@@ -0,0 +1,258 @@
+/* Lab2 of Advanced Data Structures and Algorithms (Fall 2015)
```
Comment (kmfarley11) [inline Lab2/digraph.cpp~]: These ~ files shouldn't be on our repository. They aren't doing anything so please delete them.

**Comment by kmfarley11** [comment]
The .o files and executables do not belong on our repo (for now). This is because they will keep changing and thus will muddy up our repo as well as future pull requests. Please delete them (simply run make clean) when you next push.

**Comment by kmfarley11** [comment]
I will now close this PR until the discussed issues are met. Let me know if you have any problems, otherwise issue another pull request for this when the code has been updated.

File: Lab2/digraph.cpp (hunk 1 above)
Comment (tobeja) [inline Lab2/digraph.cpp]: Not sure what's causing this, everything is lined up for me.

On Oct 13, 2015 1:54 PM, Kevin <notifications@github.com> wrote:

In Lab2/digraph.cpp<https://github.com/kmfarley11/EECE4040_2015FS/pull/20#discussion_r41900933>:

> +  string tempString;
> +
> +  edgeList* newEdge = new edgeList();
> +  newEdge->taskNo = taskTwo;
> +  newEdge->next = NULL;
> +  newEdge->index = 1;
> +
> +  //check if starting node = target node
> +  if (taskOne == taskTwo){
> +             cout << "ERROR: Edge must be between two nodes\n";
> +             return;
> +  }
> +
> +  //traverse header list
> +  while (ptr != NULL && ptr->taskNo != taskOne){
> +      if (ptr->taskNo == taskTwo)


Awkward indentations make code harder to read. In order to help me review your code please fix spacings/ indentations to increase readability. I don't mind different spacing than mine; however, your code should be more uniform looking.

Comment (kmfarley11) [inline Lab2/digraph.cpp]: Sorry, I was unclear. I was trying to make note of how lines 150 & 151 (cout << "ERROR:...") are indented quite differently from line 156 (if (ptr->taskNo...).

Also particularly note lines 162 - 165. The bracket should line up with the if, and each line has a different indentation style.

If you do not see this in your editor then it may be a tabbing issue. To fix that either replace tabs with spaces, configure tabbing sizes, or use a different editor that already has tabbing configurations. Just a note it is standard to replace tabs with 4 spaces; however i think i may have been using 2 spaces.
